app/lib/spam.js

```diff
@@ -0,0 +1,27 @@
+import { getConfig } from '#app/config'
+import { isNonEmptyArray } from '#lib/boolean_tests'
+
+let suspectKeywordsPattern
+
+async function initSuspectKeywordsPattern () {
+  if (suspectKeywordsPattern === undefined) {
+    const config = await getConfig()
+    if (isNonEmptyArray(config.spam?.suspectKeywords)) {
+      const { suspectKeywords } = config.spam
+      suspectKeywordsPattern = new RegExp(`(${suspectKeywords.join('|')})`, 'i')
+    } else {
+      suspectKeywordsPattern = null
+    }
+  }
+}
+
+export async function looksLikeSpam (str) {
+  await initSuspectKeywordsPattern()
+  if (suspectKeywordsPattern) {
+    return urlPattern.test(str) && suspectKeywordsPattern.test(str)
```

Comment:
```suggestion
    return urlPattern.test(str) || suspectKeywordsPattern.test(str)
```

Reply:
This would mean banning any URL from user profiles, which is way to restrictive imo. There are some very legit links in profile bios.

Reply:
ah that's a feature, my bad.